common/core/desktop/src/kmx/kmx_processor.cpp

```diff
@@ -353,20 +356,22 @@ int KMX_Processor::PostString(PKMX_WCHAR str, LPKEYBOARD lpkb, PKMX_WCHAR endstr
         m_actions.QueueAction(QIT_BELL, 0);
         break;
       case CODE_CONTEXT:        // copy the context to the output
-        for(q = m_miniContext; *q; q++) {
-          m_actions.QueueAction(QIT_CHAR, *q);
-        }
+      {
+        KMX_BOOL ignoreOutputKeystroke;
+        PostString(m_miniContext, lpkb, (PKMX_WCHAR) u16chr(m_miniContext, 0), &ignoreOutputKeystroke);
         break;
+      }
       case CODE_CONTEXTEX:
+      {
+        KMX_BOOL ignoreOutputKeystroke;
```

Comment:
chore: can this get declared above l. 371?

Reply:
I was intentionally declaring this variable locally because I didn't want confusion over its lifetime in this overly large function.

Reply:
Right. I was asking about having the local scope closer as in
```
if (*q) {
  temp = incxstr(q);
  KMX_BOOL ignoreOutputKeystroke;
  PostString(q, lpkb, temp, &ignoreOutputKeystroke);
}

Reply:
Ah sorry, yes, can do something like that.